src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/ContentPickerFieldSettingsDriver.cs

```diff
@@ -20,6 +20,11 @@ public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinitio
 
             await context.Updater.TryUpdateModelAsync(model, Prefix);
 
+            if (model.DisplayAllContentTypes)
+            {
+                model.DisplayedContentTypes = new string[0];
```

Comment:
```suggestion
                model.DisplayedContentTypes = Array.Empty<String>();
```

Just for your knowledge